File: 693-binary-number-with-alternating-bits/binary-number-with-alternating-bits.py
class Solution(object):
    def hasAlternatingBits(self, n):
        """
        :type n: int
        :rtype: bool
        """
        number=bin(n)[2:]
        if len(number)==0 or len(number)==1:
            return True
        odd=number[::2]
        even=number[1::2]
        if(odd[0]!=even[0]):
            resultodd=len(set(odd))==1
            resulteven=len(set(even))==1
            return resultodd and resulteven
        return False
        

File: 693-binary-number-with-alternating-bits/binary-number-with-alternating-bits.java
class Solution {
    public boolean hasAlternatingBits(int n) {
        int m=n;
        int count=0;
        while(m>0){
            m=m/2;
            count++;
        }
        int nums;
        if((n&1)!=0){
            nums=1;
        }
        else{
            nums=0;
        }
        for(int i=1;i<count;i++){
            if(nums==1){
                if((n&(1<<i))!=0){
                    return false;
                }
                nums=0;
            }
            else{
                if((n&(1<<i))==0){
                    return false;
                }
                nums=1;
            }
        }
        return true;
    }
}